tsconfig.json

```diff
@@ -6,23 +6,20 @@
     "moduleResolution": "node",
     "outDir": "./dist/",
     "sourceMap": true,
-    "noImplicitAny": false,
     "module": "esnext",
     "target": "es6",
     "jsx": "react",
-    "strictPropertyInitialization": false,
     "declaration": true,
-    "strict": false,
```

Comment:
I'm curious why we are removing these settings

Reply:
There are two reasons:
- These settings are the defaults for `.tsconfig` files, so removing them has not impact on the strictness of the Typescript checks used for the main part of the repo
- When extending `.tsconfig.yaml` in `.tsconfig.tests.yaml`, it's important to note that only ["files, include and exclude from the inheriting config file overwrite those from the base config file"](https://www.typescriptlang.org/tsconfig#extends). What this means for us is that including `strict`, `noUnusedLocals`, `noUnusedParameters` etc. in the base `tsconfig` stops us from overriding these values in the extended `tsconfig`, contrary to the intuition we usually have from OOP.

Reply:
Thank you for the detailed explanation!